lib/segment/src/index/field_index/geo_index/mmap_geo_index.rs

```diff
@@ -0,0 +1,338 @@
+use std::fs::{create_dir_all, remove_dir};
+use std::path::{Path, PathBuf};
+
+use common::types::PointOffsetType;
+use io::file_operations::{atomic_save_json, read_json};
+use memmap2::MmapMut;
+use memory::mmap_ops::{self, create_and_ensure_length};
+use serde::{Deserialize, Serialize};
+
+use super::mutable_geo_index::DynamicGeoMapIndex;
+use crate::common::mmap_bitslice_buffered_update_wrapper::MmapBitSliceBufferedUpdateWrapper;
+use crate::common::mmap_type::{MmapBitSlice, MmapSlice};
+use crate::common::operation_error::OperationResult;
+use crate::common::Flusher;
+use crate::index::field_index::geo_hash::{GeoHash, GeoHashRef, GEOHASH_MAX_LENGTH};
+use crate::index::field_index::mmap_point_to_values::MmapPointToValues;
+use crate::types::GeoPoint;
+
+const DELETED_PATH: &str = "deleted.bin";
+const COUNTS_PER_HASH: &str = "counts_per_hash.bin";
+const POINTS_MAP: &str = "points_map.bin";
+const POINTS_MAP_IDS: &str = "points_map_ids.bin";
+const CONFIG_PATH: &str = "mmap_field_index_config.json";
+
+type MmapGeoHash = [u8; GEOHASH_MAX_LENGTH + 1];
+
+#[repr(C)]
+#[derive(Clone, Debug)]
+struct Counts {
+    hash: MmapGeoHash,
+    points: u32,
+    values: u32,
+}
+
+#[repr(C)]
+#[derive(Clone, Debug)]
+struct PointKeyValue {
+    hash: MmapGeoHash,
```

Comment:
alignment?

Reply:
I removed `MmapGeoHash` and inserted here 64-bit `GeoHash` instead